### DISCUSSION TITLE
MQTT rarely connects

### DISCUSSION THREAD

**Discussion by ZhuDaHai** [arendst/Tasmota#17705] [Q&A] [answered]
Ok. Where to start.
I have various tasmota switches and sockets running 12.3.1 and 12.1.0. on 192.168.2.0/24 
There are all set up to talk MQTT to my Home Assistant on 172.27.3.4 via a VPN on 192.168.2.103
All the tasmota devices are using static IPs with 192.168.2.103 set as their Gateway (IPAddress2 192.168.2.103)
I can ping all the tasmota devices from HomeAssistant (with unimpressive times):
```
PING 192.168.2.92 (192.168.2.92) 56(84) bytes of data.
64 bytes from 192.168.2.92: icmp_seq=1 ttl=254 time=320 ms
64 bytes from 192.168.2.92: icmp_seq=2 ttl=254 time=240 ms
64 bytes from 192.168.2.92: icmp_seq=3 ttl=254 time=225 ms
64 bytes from 192.168.2.92: icmp_seq=4 ttl=254 time=273 ms

```
Ping does not appear to be enable on any of my tasmota devices on 192.168.2.0/24 to test from that end. However, testing from a Windows PC with the Gateway set to 192.168.2.103 was able to ping Home Assistant on 172.27.3.4 without any failures and similar ping times.

Rarely are any of the tasmota devices able to connect to MQTT Mosquitto broker in Home Assistant.
```
03:23:32.763 MQT: Attempting connection...
03:23:32.986 MQT: Connect failed to 172.27.3.4:1883, rc -2. Retry in 80 sec
03:24:58.721 MQT: Attempting connection...
03:24:58.933 MQT: Connect failed to 172.27.3.4:1883, rc -2. Retry in 90 sec
03:26:34.966 MQT: Attempting connection...
03:26:35.183 MQT: Connect failed to 172.27.3.4:1883, rc -2. Retry in 100 sec
```
Traceroute from Home Assistant to one of the tasmota devices shows only 2 hops:
```
traceroute to 192.168.2.92 (192.168.2.92), 30 hops max, 60 byte packets
 1  10.10.0.1 (10.10.0.1)  236.642 ms  236.530 ms  236.463 ms
 2  192.168.2.92 (192.168.2.92)  336.947 ms  336.905 ms  336.831 ms
```

They do, however, occasionally connect (at their whim), and I'm able to turn them on/off via Home Assistant.

Running MQTT Explorer in Windows and connected to Home Assistant, I never see any stat messages from any of the tasmota devices that are still trying to MQTT connect to Home Assistant.

There are no connection attempts in the Mosquitto Broker log from any of the tasmota devices that cannot connect.
Those devices that had previously connected will eventually get disconnect with a "exceeded timeout, disconnecting" error in the broker log. 

The Wifi signal levels for all these problem devices range from -58 dBm to -68 dBm, but I don't seem to have any problems connecting to their web interfaces.

I have various tasmota devices on the same network as Home Assistant (172.27.3.0/24) running 6.x.x up through 12.3.1 and only rarely have an issue connecting via MQTT MQTT to Home Assistant on 172.27.3.4, but immediately reconnect if they do.  

I can't find any timeout setting in tasmota for MQTT or any other performance settings in there. 

Any ideas where to look or what to check?

Thanks!

**Comment by arendst** [comment]
Chk commands ``MqttTimeout`` and ``MqttWifiTimeout``

**Reply by ZhuDaHai** [reply]
I totally missed that command. Sooooo embarrassed. 
Thank you!!

**Comment by barbudor** [comment, marked as answer]
My first impression is that it is essentially a network issue with whatever router you are using to bridge the 2 networks.

The ping time seems high and you may need to increase MqttWifiTimeout
See the doc as the default is 200ms way below your ping time https://tasmota.github.io/docs/Commands/#mqttwifitimeout

Now, if that works, you will know the reason
Is that a solution or a workaround?
You are talking about a VPN but you don''t tell if the sites are same or different, if different how are they connected.

I have a few devices connected to a broker over the internet (flespi.io) and I never needed to increase that value despite the devices being connected through a low end DSL

**Comment by ZhuDaHai** [comment]
Ya know, I must be BLIND. I read through that entire tasmota Commands doc before posted and COMPLETELY Missed that command. I swear I even searched for MQTT and didn't see it either. *Egg-on-face*

I set "MqttWifiTimeout 500" on every device and on all their next attempts they connected immediately!!!

The VPN are between different sites over the internet. 

I will let this run for a few days, but I'm pretty confident that was the fix!

THANK YOU!!!
